geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisSortedSet.java

```diff
@@ -323,18 +339,48 @@ public long zinterstore(RegionProvider regionProvider, RedisKey key, List<ZKeyWe
           }
         }
       }
+
       if (addToSet) {
-        memberAdd(member, newScore);
+        OrderedSetEntry entry = interMembers.get(member);
+        if (entry == null) {
+          entry = new OrderedSetEntry(member, newScore);
+          interMembers.put(member, entry);
+          interScores.add(entry);
+        } else if (entry.getScore() != newScore) {
+          interScores.remove(entry);
+          entry.updateScore(newScore);
+          interScores.add(entry);
+        }
       }
     }
 
-    if (removeFromRegion()) {
-      regionProvider.getDataRegion().remove(key);
-    } else {
-      regionProvider.getLocalDataRegion().put(key, this);
+    return sortedSetOpStoreResult(regionProvider, key, interMembers, interScores);
+  }
+
+  @VisibleForTesting
+  static long sortedSetOpStoreResult(RegionProvider regionProvider, RedisKey destinationKey,
+      MemberMap interMembers, ScoreSet interScores) {
+    RedisSortedSet destinationSet =
+        regionProvider.getTypedRedisDataElseRemove(REDIS_SORTED_SET, destinationKey, false);
+
+    if (interMembers == null || interScores.isEmpty()) {
```

Comment:
Do we want to avoid passing null because of the NPE?
Also, MemberMap has to take in a size, so far I set it as the keyweights size. Would that be the route to go or should it be set to something else? Also in doing that wold that allocate unnecessary memory or would that be negligible?

Reply:
Use of null as an argument is generally considered to be a "code smell" as outlined [here](https://medium.com/thinkster-io/code-smell-null-check-a0c4851fafbf) and [here](https://dzone.com/articles/code-smells-null). For the `MemberMap` on line 299, since we know it's empty and won't have anything added to it, just using a size of 0 would be fine.